src/ripe_rainbow/domain/base/interactions.py

```diff
@@ -135,11 +135,14 @@ def click(self, selector, text = None):
         :return: The clicked element if there's any otherwise an invalid value.
         """
 
-        element = self.waits.visible(selector, text = text, ensure = False)
-
         # waits until the try click operation is possible meaning that a
         # proper click has been "done" by the driver
         return self.waits.until(
-            lambda d: self.driver.safe(self.driver.click, element),
+            lambda d: self._click(selector, text = text),

```

Comment:
why the safe removal ?

Reply:
Moved it to the `_click` auxiliary function.